File: app/src/main/java/com/petrulak/sample/di/AppModule.java
package com.petrulak.sample.di;

import android.content.Context;
import android.net.ConnectivityManager;

import com.petrulak.sample.MyApp;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

  private MyApp appContext;

  public AppModule(MyApp app) {
    this.appContext = app;
  }

  @Singleton
  @Provides
  Context provideAppContext() {
    return appContext;
  }

  @Singleton
  @Provides
  ConnectivityManager provideConnectivityManager() {
    return (ConnectivityManager) appContext.getSystemService(Context.CONNECTIVITY_SERVICE);
  }


}

File: app/src/main/java/com/petrulak/sample/di/AppModule.kt
package com.petrulak.sample.di

import android.content.Context
import android.net.ConnectivityManager
import com.petrulak.sample.MyApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule(private val appContext: MyApp) {

  @Singleton
  @Provides
  internal fun provideAppContext(): Context {
    return appContext
  }

  @Singleton
  @Provides
  internal fun provideConnectivityManager(): ConnectivityManager {
    return appContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
  }

}
